presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentStatusUtils.java

```diff
@@ -0,0 +1,146 @@
+    private static final long WAIT_TIME_SECONDS = 60;
+    private static final long STEP_MILLISECONDS = 100;
+    private static final String LAUNCHER_SCRIPT = "/usr/lib/presto/bin/launcher";
+    private static final Path PATH_TO_CONFIG_FILE = Paths.get("/etc/presto/config.properties");
+
+    private AgentStatusUtils() {}
+
+    public static void startPresto()
+            throws PrestoManagerException
+    {
+        if (executeCommand("rpm -q presto-server-rpm") != 0) {
+            throw new PrestoManagerException(format("Presto not installed"));
+        }
+        if (executeCommand("service presto status") == 0) {
+            throw new PrestoManagerException("Presto is already running");
+        }
+        if (executeCommand("service presto start") != 0) {
+            throw new PrestoManagerException("Failed to start presto");
+        }
+        long endTime = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(WAIT_TIME_SECONDS);
+        while (System.currentTimeMillis() < endTime && !Thread.currentThread().isInterrupted()) {
+            if (executeCommand("service presto status") == 0) {
+                return;
+            }
+            else {
+                sleepUninterruptibly(STEP_MILLISECONDS, TimeUnit.MILLISECONDS);
+            }
+        }
+        throw new PrestoManagerException("Failed to start presto");
+    }
+
+    public static void stopPresto(StopType stopType)
+            throws PrestoManagerException
+    {
+        if (executeCommand("service presto status") != 0) {
+            throw new PrestoManagerException("Presto is not running");
+        }
+        switch (stopType) {
+            case TERMINATE:
+                if (executeCommand("service presto stop") != 0) {
+                    throw new PrestoManagerException("Failed to stop presto");
+                }
+                break;
+            case KILL:
+                executeCommand(format("sudo %s kill", LAUNCHER_SCRIPT));
+                if (executeCommand("service presto status") != 3) {
+                    throw new PrestoManagerException("Failed to stop presto");
+                }
+                break;
+            case GRACEFUL:
+                try {
+                    if (getFileProperty(PATH_TO_CONFIG_FILE, "coordinator").equalsIgnoreCase("true")) {
+                        throw new PrestoManagerException("Coordinator can't be gracefully stopped");
+                    }
+                    String prestoPort = getFileProperty(PATH_TO_CONFIG_FILE, "http-server.http.port");
+                    Response response = newRequestBuilder(new URI(format("localhost:%s/v1/info/state", prestoPort)))
+                            .withBody(json("SHUTTING_DOWN")).buildPut().invoke();
+                    if (response.getStatus() != 200) {
+                        throw new PrestoManagerException("Failed to stop presto gracefully");
+                    }
+                }
+                catch (IOException | URISyntaxException e) {
```

Comment:
Since you are catching this exception, I would put this URI as a parameter of the method.